central/processlisteningonport/datastore/datastore_impl_test.go

```diff
@@ -2132,3 +2142,333 @@ func (suite *PLOPDataStoreTestSuite) TestAddPodUids() {
+			pods:                  set.NewFrozenStringSet(),
+			expectedPlopDeletions: []string{fixtureconsts.PlopUID1, fixtureconsts.PlopUID2, fixtureconsts.PlopUID3, fixtureconsts.PlopUID4, fixtureconsts.PlopUID5, fixtureconsts.PlopUID6},
+		},
+		{
+			name: "deployments one missing pod - remove plops with PodUid with no matching pod even though there are deployments",
+			initialPlops: []*storage.ProcessListeningOnPortStorage{
+				fixtures.GetPlopStorage1(),
+				fixtures.GetPlopStorage2(),
+				fixtures.GetPlopStorage3(),
+				fixtures.GetPlopStorage4(),
+				fixtures.GetPlopStorage5(),
+				fixtures.GetPlopStorage6(),
+			},
+			deployments:           set.NewFrozenStringSet(fixtureconsts.Deployment6, fixtureconsts.Deployment5, fixtureconsts.Deployment3),
+			pods:                  set.NewFrozenStringSet(fixtureconsts.PodUID1, fixtureconsts.PodUID2),
+			expectedPlopDeletions: []string{fixtureconsts.PlopUID6},
+		},
+		{
+			name: "one missing deployments no missing pods - remove plops with no matching deployments even though there are matching pods",
+			initialPlops: []*storage.ProcessListeningOnPortStorage{
+				fixtures.GetPlopStorage1(),
+				fixtures.GetPlopStorage2(),
+				fixtures.GetPlopStorage3(),
+				fixtures.GetPlopStorage4(),
+				fixtures.GetPlopStorage5(),
+				fixtures.GetPlopStorage6(),
+			},
+			deployments:           set.NewFrozenStringSet(fixtureconsts.Deployment5, fixtureconsts.Deployment3),
+			pods:                  set.NewFrozenStringSet(fixtureconsts.PodUID1, fixtureconsts.PodUID2, fixtureconsts.PodUID3),
+			expectedPlopDeletions: []string{fixtureconsts.PlopUID1, fixtureconsts.PlopUID4},
+		},
+		{
+			name: "no missing deployments or pods but plops are expired - remove all expired plops",
+			initialPlops: []*storage.ProcessListeningOnPortStorage{
+				fixtures.GetPlopStorage4(),
+				fixtures.GetPlopStorageExpired1(),
+				fixtures.GetPlopStorageExpired2(),
+				fixtures.GetPlopStorageExpired3(),
+			},
+			deployments:           set.NewFrozenStringSet(fixtureconsts.Deployment6, fixtureconsts.Deployment5, fixtureconsts.Deployment3),
+			pods:                  set.NewFrozenStringSet(fixtureconsts.PodUID1, fixtureconsts.PodUID2, fixtureconsts.PodUID3),
+			expectedPlopDeletions: []string{fixtureconsts.PlopUID7, fixtureconsts.PlopUID8, fixtureconsts.PlopUID9},
+		},
+	}
+	for _, c := range cases {
+		suite.T().Run(c.name, func(t *testing.T) {
+			suite.TearDownTest()
+			suite.SetupTest()
+			// Add deployments if necessary
+			deploymentDS, err := deploymentStore.GetTestPostgresDataStore(suite.T(), suite.postgres.DB)
+			suite.Nil(err)
+			for _, deploymentID := range c.deployments.AsSlice() {
+				suite.NoError(deploymentDS.UpsertDeployment(suite.hasAllCtx, &storage.Deployment{Id: deploymentID, ClusterId: fixtureconsts.Cluster1}))
+			}
+
+			for _, podID := range c.pods.AsSlice() {
+				insertPod := fmt.Sprintf("INSERT INTO pods (id, clusterid) VALUES ('%s', '%s')", podID, fixtureconsts.Cluster1)
```

Comment:
It was not possible to use the pod store here as that would have created cyclical imports.